src/main/scala/org/graphframes/lib/ConnectedComponents.scala

```diff
@@ -346,9 +340,14 @@ private object ConnectedComponents extends Logging {
       ee.persist(StorageLevel.MEMORY_AND_DISK)
 
       // test convergence
-      val currSum = ee.select(sum(col(SRC))).as[Long].first()
+
+      // Taking the sum in DecimalType to preserve precision.
+      // We use 20 digits for long values and Spark SQL will add 10 digits for the sum.
+      // It should be able to handle 100 billion edges without overflow.
```

Comment:
Ok then I think we should either document that this can fail silently for > 100 billion edges or explicitly check and throw an exception.